src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -879,6 +879,9 @@ public static bool SequenceEqual<T>(this IEnumerable<T>? first, IEnumerable<T>?
             }
         }
 
+        // https://github.com/dotnet/runtime/issues/107723
+        public static IEnumerable<T> Reverse<T>(this T[] source) => Enumerable.Reverse(source);
```

Comment:
It's probably not correct to put this here. @jaredpar to confirm, but I'm concerned that we'll end up breaking ourselves when the bcl adds this.

Reply:
https://github.com/dotnet/roslyn/blob/5a39d0ad691ee88dcf163808d9ed7a46b168a1c9/docs/contributing/Target%20Framework%20Strategy.md#pattern-for-extension-methods

I think this should be fine. Problems could arise if we wrap this in `#if`s incorrectly (so when bcl adds this we need to be careful making this conditional).

Reply:
I _think_ that we should be able to mitigate this by wrapping this in 

```csharp
#if !NET10_OR_GREATER
```

By the time that we actually start targeting `net10.0` this API is going to be there so that won't cause a break on us. We generally move to the new TFM rather late in the release and think that we can expect this to be added early given it's tied to first class span support which is going in early.

Reply:
Won't we potentially break in a torn dll scenario though? Or have we thoroughly removed any change of that at this point?

Reply:
What teared scenario are you thinking of? This is just another API inside the compiler. The only impact that can happen is weirdness at compile time. The `#if` will ensure that `net9.0` or earlier we have the method. If we get deployed into `net10.0` or higher then there is just another polyfill API in our binary. In the case the `#if` is `false` then we're in `net10.0` where the API is guaranteed to be available.

Reply:
Well, we certainly had a tear like this when we had the EA dll. Have we gotten rid of all of those scenarios?

Reply:
Shouldn't have this type of tearing problem without the EA DLL

Reply:
Hm, good idea about adding the `#if` for .NET 10, I have done that. Now it matches our guidelines linked above precisely.